spidermon/contrib/scrapy/monitors.py

```diff
@@ -495,6 +498,31 @@ def test_check_if_field_coverage_rules_are_met(self):
         self.assertTrue(len(failures) == 0, msg=msg)
 
 
+@monitors.name("Periodic execution time monitor")
+class PeriodicExecutionTimeMonitor(Monitor, StatsMonitorMixin):
+    """Check for runtime exceeding a target maximum runtime.
+
+    You can configure the maximum runtime (in seconds) using
+    ``SPIDERMON_MAX_EXECUTION_TIME`` as a project setting or spider attribute."""
+
+    @monitors.name("Maximum execution time reached")
+    def test_execution_time(self):
+        crawler = self.data.get("crawler")
+        max_execution_time = crawler.settings.getint(SPIDERMON_MAX_EXECUTION_TIME)
+        if not max_execution_time:
+            return
+        now = datetime.datetime.now()
+        start_time = self.data.stats.get("start_time")
```

Comment:
**question:** I believe start_time is already a `datetime` object? At least that's what it looks like in the stats when running locally.
```
2022-01-04 14:22:34 [scrapy.statscollectors] INFO: Dumping Scrapy stats:
{'elapsed_time_seconds': 0.015785,
 'finish_reason': 'finished',
 'finish_time': datetime.datetime(2022, 1, 4, 17, 22, 34, 348831),
 'log_count/DEBUG': 1,
 'log_count/ERROR': 1,
 'log_count/INFO': 11,
 'memusage/max': 134189056,
 'memusage/startup': 134189056,
 'start_time': datetime.datetime(2022, 1, 4, 17, 22, 34, 333046)}
```

Reply:
> **question:** I believe start_time is already a `datetime` object? At least that's what it looks like in the stats when running locally.

Ah yes, the zyte dash stats tab displays it as a unix timestamp but I can see that it was converted from a datetime when the spider was finished. Thanks for calling this out!